mgr/dashboard: User Management E2E tests

adk3798 opened the pull request:
Test breadcrumbs for user management pages
Test creating, editing and deleting a user
Test creating, editing and deleting a role

Fixes: https://tracker.ceph.com/issues/41231
Fixes: https://tracker.ceph.com/issues/41232
Fixes: https://tracker.ceph.com/issues/41233

Signed-off-by: Adam King <adking@redhat.com>
Signed-off-by: Rafael Quintero <rquinter@redhat.com>

adk3798 commented:
jenkins test dashboard

adk3798 commented:
jenkins test dashboard

tspmelo commented:
jenkins test dashboard

votdev commented:
Can you please rebase this PR, i stopped testing this PR after manual merging various conflicts in src/rgw more than 8 times.

tspmelo commented on src/pybind/mgr/dashboard/frontend/e2e/user-mgmt.po.ts:
```diff
@@ -0,0 +1,118 @@
+    // Click the create button and wait for user to be made
+    const createButton = element(by.cssContainingText('button', 'Create User'));
+    this.moveClick(createButton).then(() => {
+      browser.wait(Helper.EC.presenceOf(this.getTableCell(username)), 10000);
```
Same for other occurrences. 
```suggestion
      browser.wait(Helper.EC.presenceOf(this.getTableCell(username)), Helper.TIMEOUT);
```

tspmelo commented on src/pybind/mgr/dashboard/frontend/e2e/user-mgmt.po.ts:
```diff
@@ -0,0 +1,118 @@
+  pages = {
+    index: '/#/user-management',
+    users: '/#/user-management/users',
+    userscreate: '/#/user-management/users/create',
```
Easier to read:
```suggestion
    userCreate: '/#/user-management/users/create',
```

tspmelo commented on src/pybind/mgr/dashboard/frontend/e2e/user-mgmt.po.ts:
```diff
@@ -0,0 +1,118 @@
+    users: '/#/user-management/users',
+    userscreate: '/#/user-management/users/create',
+    roles: '/#/user-management/roles',
+    rolescreate: '/#/user-management/roles/create'
```
```suggestion
    roleCreate: '/#/user-management/roles/create'
```

adk3798 commented:
Updates:
 - Rebased
 - Changed wait times to Helper.TIMEOUT
 - changed page url variables 'userscreate' and 'rolescreate' to 'userCreate' and 'roleCreate' respectively. Should be easier to read

adk3798 commented:
@votdev I tried to rebase. Is this working for you now?

adk3798 commented:
Made some mistakes when updating. Going to fix

adk3798 commented:
jenkins retest this please

adk3798 commented:
jenkins retest this please

adk3798 commented:
jenkins test make check

adk3798 commented:
jenkins test make check

tspmelo commented:
Please run "npm run fix"

adk3798 commented:
Fixed linting error that was causing make check to fail
